src/troute-network/troute/DataAssimilation.py

```diff
@@ -238,10 +238,12 @@ def update_after_compute(self, run_results, time_increment):
             - lastobs_df               (DataFrame): Last gage observations data for DA
         '''
         streamflow_da_parameters = self._data_assimilation_parameters.get('streamflow_da', None)
- 
+
         if streamflow_da_parameters:
             if streamflow_da_parameters.get('streamflow_nudging', False):
-                self._last_obs_df = new_lastobs(run_results, time_increment)
+                if not run_results[0][3][0].size == 0:
```

Comment:
I think this check needs to be modified. This only checks if the first object in `run_results` has no gage information. But this varies for each object:

```
(Pdb) [rr[3][0].size for rr in run_results]
[2,2,0,2,0,1,0,0,3,0,17,1,1,4,3,1,0,0,0,2,0,1,2,0,0,0,1,0,3,1,2,1,1,0,1,1,0,1,0,1,3,0,0,0,0,0,0,0,3,1,0,11,0,2,0,0,3,1,0,0,0,2,4,0,0,3,0,0,0,0]
(Pdb)
```

Reply:
what about checking self._last_obs_df? If it is empty in the first place, then we expect nothing will come out for lastobs values from mc_reach.pyx.

Reply:
Thanks for your suggestion.